.github/workflows/gpuci.yml

```diff
@@ -0,0 +1,41 @@
+name: "GPU CI/CD"
+
+on:
+  pull_request:
+    branches:
+      - 'main'
+      - 'r**'
+    types: [ labeled ]
+
+jobs:
+  gpu-test:
+    runs-on: self-hosted
+    if: ${{ github.event.label.name == 'gpuci' }}
+
+    steps:
+    - name: Run nvidia-smi test
+      run: |
+        whoami
+        nvidia-smi
+
+    - name: Checkout the PR code
+      uses: actions/checkout@v3
+
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v4
```

Comment:
Hey folks!

Are you planning to migrate to a containerized test later? With our non-ephemeral infra I have a bit of worries with pure python installs. This can lead to nasty unexpected behavior. Workflows don’t run in isolation, thus concurrently running workflows can override each others environment. A containerized workflow (either building freshly or inheriting from NeMo) would be much safer & easier to debug.